browser_history/generic.py

```diff
@@ -84,20 +86,20 @@ def __init__(self, plat: utils.Platform = None):
 
     def profiles(self) -> typing.List[str]:
         """Returns a list of profile directories
-
-        (TODO: fix this)
-        The returned profiles include only the final name in the path.
-
         :rtype: list(str)
         """
         if not self.profile_support:
-            return ['.']
-        maybe_profile_dirs = []
-        for profile_dir_prefix in self.profile_dir_prefixes:
-            maybe_profile_dirs.extend(self.history_dir.glob(profile_dir_prefix))
-        profile_dirs = [profile_dir.name for profile_dir in maybe_profile_dirs
-                        if (profile_dir / self.history_file).exists()]
-
+            return ["."]
+        file = glob.glob(
+            str(self.history_dir) + "/**/" + str(self.history_file), recursive=True
+        )
+        profile_dirs = []
+        for profile_dir in file:
+            profile_dirs.append(
+                profile_dir.split(str(self.history_dir) + "/", maxsplit=2)[-1].split(
+                    "/" + self.history_file, maxsplit=2
```

Comment:
Here the `maxsplit` should be 1, right? Since we need exactly 2 items to be returned.

Reply:
ohh yeahh, I got confused  , I'll change it right away

Reply:
done